OTAnalytics/plugin_parser/track_export.py

```diff
@@ -43,11 +43,14 @@ def _get_data(self) -> DataFrame:
         dataset = self._track_repository.get_all()
         if isinstance(dataset, PandasDataFrameProvider):
             return dataset.get_data().reset_index()
-        detections = [
-            [detection.to_dict() for detection in track.detections]
-            for track in dataset.as_list()
-        ]
-        return DataFrame.from_dict(detections)
+        detections = []
+        for _track in dataset.as_list():
+            track_classification = _track.classification
+            for detection in _track.detections:
+                current = detection.to_dict()
+                current[track.TRACK_CLASSIFICATION] = track_classification
```

Comment:
Why do we need to change the track_classification for each detection?
Is it incorrect in the dataframe?

Reply:
We are actually not changing `track_classification`, but adding it to the detection dictionary, since a `Detection` object does not have a member called track_classification. The information about the `track_classification` is only contained in the `Track` class.